src/pyfmi/simulation/assimulo_interface_fmi3.pyx

```diff
@@ -444,29 +443,29 @@ cdef class FMIODE3(cExplicit_Problem):
         #         raise FMUException("Failed logging diagnostics, number of data points expected to be {} but was {}".format(self._number_of_diagnostics_variables, index))
         #     self.export.diagnostics_point(diag_data)
 
-        # # Enter event mode
-        # self._model.enter_event_mode()
+        # Enter event mode
+        self._model.enter_event_mode()
 
-        # self._model.event_update()
-        # eInfo = self._model.get_event_info()
+        self._model.event_update()
+        eInfo = self._model.get_event_info()
 
-        # # Check if the event affected the state values and if so sets them
-        # if eInfo.valuesOfContinuousStatesChanged:
-        #     if self._extra_f_nbr > 0:
-        #         solver.y = self._model.continuous_states.append(solver.y[-self._extra_f_nbr:])
-        #     else:
-        #         solver.y = self._model.continuous_states
+        # Check if the event affected the state values and if so sets them
+        if eInfo.valuesOfContinuousStatesChanged:
+            if self._extra_f_nbr > 0:
+                solver.y = self._model.continuous_states.append(solver.y[-self._extra_f_nbr:])
+            else:
+                solver.y = self._model.continuous_states
 
         # # Get new nominal values.
-        # if eInfo.nominalsOfContinuousStatesChanged:
+        # if eInfo.nominalsOfContinuousStatesChanged: # TODO
         #     solver.atol = 0.01*solver.rtol*self._model.nominal_continuous_states
```

Comment:
This doesn't seem quite right, since this could overwrite other rtol vs. atol ratios set at the start of a simulation, I propose to take this later though.